x/twap/api.go

```diff
@@ -41,12 +41,49 @@ func (k Keeper) GetArithmeticTwap(
 	quoteAssetDenom string,
 	startTime time.Time,
 	endTime time.Time,
+) (sdk.Dec, error) {
+	arithmeticStrategy := &arithmetic{k}
+	return k.getTwap(ctx, poolId, baseAssetDenom, quoteAssetDenom, startTime, endTime, arithmeticStrategy)
+}
+
+// GetArithmeticTwapToNow returns GetArithmeticTwap on the input, with endTime being fixed to ctx.BlockTime()
+// This function does not mutate records.
+func (k Keeper) GetArithmeticTwapToNow(
+	ctx sdk.Context,
+	poolId uint64,
+	baseAssetDenom string,
+	quoteAssetDenom string,
+	startTime time.Time,
+) (sdk.Dec, error) {
+	arithmeticStrategy := &arithmetic{k}
+	return k.getTwapToNow(ctx, poolId, baseAssetDenom, quoteAssetDenom, startTime, arithmeticStrategy)
+}
+
+// GetBeginBlockAccumulatorRecord returns a TwapRecord struct corresponding to the state of pool `poolId`
+// as of the beginning of the block this is called on.
+// This uses the state of the beginning of the block, as if there were swaps since the block has started,
+// these swaps have had no time to be arbitraged back.
+// This accumulator can be stored, to compute wider ranged twaps.
```

Comment:
I'm not sure I'm following everything described here. I think the method is as simple as returning a most recent record with accumulator values interpolated to the current block time.

Reply:
hmm, not sure about the nuances of this function as this existed before this PR. I can remove it if u suggest

Reply:
I suggest removing it because it felt misleading to me. I think we should focus on the most basic functionality of what the function does in its spec

Reply:
gotchu will do